udffsck/options.c

```diff
@@ -0,0 +1,192 @@
+    {"autofix",    no_argument, 0, 'p'},
+    {"check", no_argument, 0, 'c'},
+    {"colors",    no_argument,       0, 'C'},
+    {"fast",    no_argument,       0, 'f'},
+    {"help",    no_argument,       0, 'h'},
+    {0, 0, 0, 0}
+};
+
+/**
+ * Help string for options
+ */
+static char * help[] = {
+    "Increase verbosity. Without it are printed only error messages, -v prints warnings, -vv is for humans, -vvv is for developers and curious people.",
+    "Medium block size. Mandatory parameter, can be 512, 1024, 2048 or 4096.",
+    "Medium is will be fixed interactivelly and all fixings must be authorized by user.",
+    "Medium is will be fixed automatically. All found errors will be fixed if possible.",
+    "Medium will be only checked. This is default behavior, but this flag override -p.",
+    "Tool output will be colored with ASCII color codes.",
+    "Fast mode: File tree check will be skipped.",
+    "This help message.",
+    ""
+}; 
+
+/**
+ * \brief Prints out usage
+ */
+void usage(void)
+{
+    int i;
+
+    printf("udffsck " UDFFSCK_VERSION  " from " PACKAGE_NAME " " PACKAGE_VERSION ".");
+    printf("\nUsage:\n\tudffsck [-icpvvvCh] [-B blocksize] medium\n");
+    printf("Options:\n");
+    for (i = 0; long_options[i].name != NULL; i++) {
+        if (long_options[i].flag != 0)
+            printf("  --%s\t", long_options[i].name);
+        else
+            printf("  -%c\t", long_options[i].val);
+        printf(" %s\n", help[i]);
+    }
+    printf("Return codes:\n");
+    printf("  0 - No error\n"
+           "  1 - Filesystem errors were fixed\n"
+          /* "  2 - Filesystem errors were fixed, reboot is recomended\n"*/
+           "  4 - Filesystem errors remained unfixed\n"
+           "  8 - Program error\n"
+           "  16 - Wrong input parameters\n"
+           "  32 - Check was interrupted by user request\n"
+          /* "  128 - Shared library error"*/
+           "\n");
+    exit(32);
+}
+
+/**
+ * \brief Input argument parsing function
+ *
+ * This function parse thru inputs. It recognizes following: -vvvipch -B <BLOCKSIZE>. Without any argument is only path to medium.
+ */
+void parse_args(int argc, char *argv[], char **path, int *blocksize) 
+{
+    int c;
+
+    while (1)
+    {
+        /* getopt_long stores the option index here. */
+        int option_index = 0;
+
+        c = getopt_long (argc, argv, "vB:ipcCfh", long_options, &option_index);
+
+        /* Detect the end of the options. */
+        if (c == -1)
+            break;
+
+        switch (c)
+        {
+            case 0:
+                /* If this option set a flag, do nothing else now. */
+                if (long_options[option_index].flag != 0)
+                    break;
+                printf ("option %s", long_options[option_index].name);
+                if (optarg)
+                    printf (" with arg %s", optarg);
+                printf ("\n");
+                break;
+
+            case 'B':
```

Comment:
use small b for consistency with other udf tools.

Reply:
Fixed here:  b31664